data-filter.html

```diff
@@ -741,6 +776,46 @@ <h3 style="text-align: center;">Selected Measurements</h3>
                 this._updateUnits();
             }
 
+            _maxAmountOfMeasurements() {
+              return Math.max(...Object.values(this._attributesGroupedBy).map(it => it.length));
+            }
+
+            _attributesGroupedByObserver(newVal, oldVal) {
+              const max = this._maxAmountOfMeasurements();
+              if (max > 0 && this.selectionUnit.unit === '#') {
+                this.set('selectionUnit.max', max);
+                this.notifyPath('selectionUnit.max');
+              }
+            }
+
+            _computeAttributesGroupedBy(_attributes, _groupedBy, data) {
+              const groupedByAnnotation = {};
+                if (this._groupedBy !== 0) {
+                  const annotationType = this._attributes[this._groupedBy - 1];
+                  data.reduce((acc, attr) => {
+                    if(!acc[this._getSanitizeItemName(attr[annotationType.name])]) {
+                      acc[this._getSanitizeItemName(attr[annotationType.name])] = [];
+                    }
+                    acc[this._getSanitizeItemName(attr[annotationType.name])].push(attr);
+                    return acc;
+                  }, groupedByAnnotation);
```

Comment:
theres a bug with the selection # & how many it shows

your input to the drop down for `groupby` is only restricted to categorical attributes. Line 432 calls `_getAttributes`
Theres a mismap of index between the filtered list and the original `_attributes` list. you cannot assume they both will contain the same number of elements -

Imagine a case where a collection has 10 attributes, but the first 9 are numerical and only the last one is categorical, when i now choose to `groupby` by this field, `this._groupedBy` will be 2, but in the `_attributes` field, its index is 9.